### PR TITLE
Update sampler in server configuration

### PR DESCRIPTION
**Pull request by ProKil**
Closes # 

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [x] All the tests have passed
- [x] Branch name follows `type/descript` (e.g. `feature/add-llm-agents`)
- [x] Ready for code review

### CODE REVIEW

File: sotopia_conf/server_conf/server.gin [hunk 1]
```diff
@@ -14,3 +15,4 @@ server.run_async_server:
     using_async=True
     tag=%TAG
     omniscient=%OMNISCIENT
+    sampler=@samplers.UniformSampler()
```
Comment (XuhuiZhou) [inline sotopia_conf/server_conf/server.gin:18]: make this a gin parameter to support customization.

Comment (XuhuiZhou) [inline sotopia_conf/server_conf/server.gin:18]: make this a gin parameter to support customization.

**Review by XuhuiZhou — CHANGES_REQUESTED** [review]

File: sotopia_conf/server_conf/server.gin (hunk 1 above)
Comment (ProKil) [reply sotopia_conf/server_conf/server.gin:18]: No this is not possible. You will get 
```
ValueError: No configurable matching reference '@sotopia.samplers.BaseSampler()'.
  In bindings string line 1
    SAMPLER = @sotopia.samplers.BaseSampler()
```
after setting the gin parameter

Comment (ProKil) [reply sotopia_conf/server_conf/server.gin:18]: The reason is that you have to import sotopia in the gin configuration. This is not possible from command line arguments, and should not be recommended to our users.

**Comment by XuhuiZhou** [comment]
Then how can we choose to use different samplers in an easy way? I think it
does not make sense to bury the hyper parameter that deep?

**Comment by ProKil** [comment]
I know what you mean, let me add an example of using sotopia.server.run_async_server directly. From the perspective of the `sotopia` package, it is actually well exposed. `sotopia_conf` is merely a recommended way to config the experiments.

**Review by XuhuiZhou — APPROVED** [review]

**Pull request merged**